### PR TITLE
Add certificate rotation integration tests

### PR DESCRIPTION
**Pull request by galal-hussein**
#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: tests/integration/certrotation/certrotation_int_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		Expect(testutil.K3sKillServer(server)).To(Succeed())
+		Expect(testutil.K3sCleanup(testLock, "")).To(Succeed())
+		Expect(testutil.K3sKillServer(server2)).To(Succeed())
+		Expect(testutil.K3sCleanup(testLock, tmpdDataDir)).To(Succeed())
+	}
+})
+
+func Test_IntegrationEtcdSnapshot(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Etcd Snapshot Suite")
```
Comment (dereknola) [inline tests/integration/certrotation/certrotation_int_test.go:90]:
```suggestion
func Test_IntegrationCertRotation(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "Cert Rotation Suite")
```

File: tests/integration/certrotation/certrotation_int_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package snapshot_test
```
Comment (dereknola) [inline tests/integration/certrotation/certrotation_int_test.go:1]:
```suggestion
package cert_rotation_test
```

**Review by dereknola — CHANGES_REQUESTED** [review]

File: tests/integration/certrotation/certrotation_int_test.go (hunk 2 above)
Comment (galal-hussein) [reply tests/integration/certrotation/certrotation_int_test.go:1]: will fix

File: tests/integration/certrotation/certrotation_int_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	"strings"
+	"testing"
+
+	testutil "github.com/k3s-io/k3s/tests/integration"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var server, server2 *testutil.K3sServer
+var tmpdDataDir = "/tmp/certrotationtest"
```
Comment (briandowns) [inline tests/integration/certrotation/certrotation_int_test.go:13]:
```suggestion
const tmpdDataDir = "/tmp/certrotationtest"
```

**Review by dereknola — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Pull request merged**